packages/react-native-session-replay/ios/Sources/RCTTextViewRecorder.swift

```diff
@@ -0,0 +1,145 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+@_spi(Internal)
+import DatadogSessionReplay
+import React
+
+internal class RCTTextViewRecorder: SessionReplayNodeRecorder {
+    internal var textObfuscator: (SessionReplayViewTreeRecordingContext) -> SessionReplayTextObfuscating = { context in
+        return context.recorder.privacy.staticTextObfuscator
+    }
+
+    internal var identifier = UUID()
+
+    internal let uiManager: RCTUIManager
+
+    internal init(uiManager: RCTUIManager) {
+        self.uiManager = uiManager
+    }
+
+    internal func extractTextFromSubViews(
+        subviews: [RCTShadowView]?
+    ) -> String? {
+        if let subviews = subviews {
+            return subviews.compactMap { subview in
+                if let sub = subview as? RCTRawTextShadowView {
+                    return sub.text
+                }
+                if let sub = subview as? RCTVirtualTextShadowView {
+                    // We recursively get all subviews for nested Text components
+                    return extractTextFromSubViews(subviews: sub.reactSubviews())
+                }
+                return nil
+            }.joined()
+        }
+        return nil
+    }
+
+    public func semantics(
+        of view: UIView,
+        with attributes: SessionReplayViewAttributes,
+        in context: SessionReplayViewTreeRecordingContext
+    ) -> SessionReplayNodeSemantics? {
+        guard let textView = view as? RCTTextView else {
+            return nil
+        }
+
+        var shadowView: RCTTextShadowView? = nil
+        let tag = textView.reactTag
+
+        RCTGetUIManagerQueue().sync {
+            shadowView = uiManager.shadowView(forReactTag: tag) as? RCTTextShadowView
+        }
+
+        if let shadow = shadowView {
+            // TODO: RUM-2173 check performance is ok
+            let text = extractTextFromSubViews(
+                subviews: shadow.reactSubviews()
+            )
+
+            let builder = RCTTextViewWireframesBuilder(
+                wireframeID: context.ids.nodeID(view: textView, nodeRecorder: self),
+                attributes: attributes,
+                text: text,
+                textAlignment: shadow.textAttributes.alignment,
+                textColor: shadow.textAttributes.foregroundColor?.cgColor,
+                textObfuscator: textObfuscator(context),
+                font: shadow.textAttributes.effectiveFont(), // Custom fonts are currently not supported for iOS
```

Comment:
It actually goes to the player level.

Is font family output by RN web-safe? If no - there's work to be done on RN side as well.

Reply:
That's a good question, I've not looked into it yet. Usually you add font files, named whatever you want, with 1 file per variant (weight + style).
So I'd say it's the same as what is done for iOS, not web safe

Reply:
Alright!

Would be great to add a ticket for RN that's blocked by the iOS one